cmd/resim/commands/batch.go

```diff
@@ -346,6 +351,93 @@ func createBatch(ccmd *cobra.Command, args []string) {
 	}
 }
 
+func batchToSlackWebhookPayload(batch *api.Batch) *slack.WebhookMessage {
+	var (
+		suite   *api.TestSuite
+		system  *api.System
+		reports *[]api.Report
+	)
+	// todo: matt: programatically convert api url from cobra to frontend
+	baseUrl := fmt.Sprintf("%sprojects/%s", "https://app.resim.io/", batch.ProjectID.String())
+	blocks := &slack.Blocks{BlockSet: make([]slack.Block, 3, 3)}
+
+	// Get the suite object
+	suiteResponse, err := Client.GetTestSuiteWithResponse(context.Background(), *batch.ProjectID, *batch.TestSuiteID)
+	if err != nil {
+		log.Fatal("unable to retrieve suite for batch:", err)
+	}
+	ValidateResponse(http.StatusOK, "unable to retrieve suite for batch", suiteResponse.HTTPResponse, suiteResponse.Body)
+	suite = suiteResponse.JSON200
+
+	// Get the system object
+	systemResponse, err := Client.GetSystemWithResponse(context.Background(), *batch.ProjectID, *batch.SystemID)
+	if err != nil {
+		log.Fatal("unable to retrieve system for batch:", err)
+	}
+	ValidateResponse(http.StatusOK, "unable to retrieve system for batch", systemResponse.HTTPResponse, systemResponse.Body)
+	system = systemResponse.JSON200
+
+	// Intro text
+	introData := struct {
+		SuiteUrl   string
+		SuiteName  string
+		BatchUrl   string
+		SystemUrl  string
+		SystemName string
+	}{
+		fmt.Sprintf("%s/test-suites/%s/revisions/%d", baseUrl, batch.TestSuiteID.String(), *batch.TestSuiteRevision),
+		suite.Name,
+		fmt.Sprintf("%s/batches/%s", baseUrl, batch.BatchID.String()),
+		fmt.Sprintf("%s/systems/%s", baseUrl, batch.SystemID.String()),
+		system.Name,
+	}
+	introTemplate := template.Must(template.New("intro").Parse("Last night’s <{{.SuiteUrl}}|{{.SuiteName}}> *<{{.BatchUrl}}|run>* for <{{.SystemUrl}}|{{.SystemName}}> ran successfully with the following breakdown:"))
+	var introBuffer bytes.Buffer
+	introTemplate.Execute(&introBuffer, introData)
+	introTextBlock := slack.NewTextBlockObject("mrkdwn", introBuffer.String(), false, false)
+	blocks.BlockSet[0] = slack.NewSectionBlock(introTextBlock, nil, nil)
+
+	// List section
+	boldStyle := slack.RichTextSectionTextStyle{Bold: true}
+	buildListElement := func(count int, label string, filter string) *slack.RichTextSection {
+		return slack.NewRichTextSection(
+			slack.NewRichTextSectionTextElement(fmt.Sprintf("%d ", count), nil),
+			slack.NewRichTextSectionLinkElement(introData.BatchUrl+"?performanceFilter="+filter, label, &boldStyle),
+		)
+	}
+
+	listBlock := slack.NewRichTextList("bullet", 0,
+		slack.NewRichTextSection(slack.NewRichTextSectionTextElement(fmt.Sprintf("%d total tests", *batch.TotalJobs), nil)),
+		// Passed calculation borrowed from bff: https://github.com/resim-ai/rerun/blob/ebf0cde9472f555ae099e08e512ed4a7dfdf01f4/bff/lib/bff/batches/conflated_status_counts.ex#L49
+		buildListElement(batch.JobStatusCounts.Succeeded-(batch.JobMetricsStatusCounts.FailBlock+batch.JobMetricsStatusCounts.FailWarn), "Passed", "Passed"),
```

Comment:
yep, that makes sense to me: conflates N/A etc with passed. I'm sure there's some Five AI trauma there for you